File: accsaber-database/src/main/java/de/ixsen/accsaber/database/repositories/model/ScoreDataHistoryRepository.java
package de.ixsen.accsaber.database.repositories.model;

import de.ixsen.accsaber.database.model.players.PlayerData;
import de.ixsen.accsaber.database.model.players.ScoreDataHistory;
import de.ixsen.accsaber.database.repositories.ReadOnlyRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface ScoreDataHistoryRepository extends ReadOnlyRepository<ScoreDataHistory, Long, ScoreDataHistory> {

    List<ScoreDataHistory> findAllByPlayerAndLeaderboardId(PlayerData playerData, Long scoreId);
}

File: accsaber-database/src/main/kotlin/de/ixsen/accsaber/database/repositories/model/ScoreDataHistoryRepository.kt
package de.ixsen.accsaber.database.repositories.model

import de.ixsen.accsaber.database.model.players.PlayerData
import de.ixsen.accsaber.database.model.players.ScoreDataHistory
import de.ixsen.accsaber.database.repositories.ReadOnlyRepository
import org.springframework.stereotype.Repository

@Repository
interface ScoreDataHistoryRepository : ReadOnlyRepository<ScoreDataHistory, Long, ScoreDataHistory> {
    fun findAllByPlayerAndLeaderboardId(playerData: PlayerData, scoreId: Long): List<ScoreDataHistory>
}